frontend/src/components/CreateVault.vue

```diff
@@ -1,42 +1,82 @@
 <template>
-  <div v-if="state == State.Initial || state == State.Processing">
-    <form ref="form" class="space-y-6" novalidate @submit.prevent="createVault()">
+  <div v-if="state == State.Initial">
+    {{ t('common.loading') }}
+  </div>
+
+  <div v-else-if="state == State.EnterRecoveryKey">
+    <form ref="form" novalidate @submit.prevent="validateRecoveryKey()">
+      <div class="flex justify-center">
+        <div class="bg-white px-4 py-5 shadow sm:rounded-lg sm:p-6 text-center sm:max-w-lg">
+          <div class="mx-auto flex items-center justify-center h-12 w-12 rounded-full bg-emerald-100">
+            <ArrowPathIcon class="h-6 w-6 text-emerald-600" aria-hidden="true" />
+          </div>
+          <div class="mt-3 sm:mt-5">
+            <h3 class="text-lg leading-6 font-medium text-gray-900">
+              {{ t('createVault.enterRecoveryKey.title') }}
+            </h3>
+            <div class="mt-2">
+              <p class="text-sm text-gray-500">
+                {{ t('createVault.enterRecoveryKey.description') }}
+              </p>
+            </div>
+          </div>
+          <div class="mt-5 sm:mt-6">
+            <label for="recoveryKey" class="sr-only">{{ t('createVault.enterRecoveryKey.recoveryKey') }}</label>
+            <textarea id="recoveryKey" v-model="recoveryKey" rows="6" name="recoveryKey" class="block w-full rounded-md border-gray-300 shadow-sm focus:border-primary focus:ring-primary sm:text-sm" :class="{ 'invalid:border-red-300 invalid:text-red-900 focus:invalid:ring-red-500 focus:invalid:border-red-500': onRecoverError instanceof FormValidationFailedError }" required />
+          </div>
+          <div class="mt-5 sm:mt-6">
+            <button type="submit" :disabled="processing" class="inline-flex w-full justify-center rounded-md border border-transparent bg-primary px-4 py-2 text-base font-medium text-white shadow-sm hover:bg-primary-d1 focus:outline-none focus:ring-2 focus:primary focus:ring-offset-2 sm:text-sm disabled:opacity-50 disabled:hover:bg-primary disabled:cursor-not-allowed">
+              {{ t('createVault.enterRecoveryKey.submit') }}
+            </button>
+            <div v-if="onRecoverError != null">
+              <p v-if="onRecoverError instanceof FormValidationFailedError" class="text-sm text-red-900 mt-2">{{ t('createVault.error.formValidationFailed') }}</p>
+              <p v-else class="text-sm text-red-900 mt-2">{{ t('createVault.error.invalidRecoveryKey') }}</p>
+            </div>
+          </div>
+        </div>
+      </div>
+    </form>
+  </div>
+
+  <div v-else-if="state == State.EnterVaultDetails">
+    <form ref="form" class="space-y-6" novalidate @submit.prevent="validateVaultDetails()">
```

Comment:
Is it okay to bind different HTML forms to the same Vue ref?

Reply:
In this case, it's not an issue because of the `v-if`s so that the Vue ref is always only used "once".